gemfirexd/core/src/main/java/com/pivotal/gemfirexd/internal/impl/sql/compile/ResultColumnList.java

```diff
@@ -4113,6 +4115,14 @@ public void checkAutoincrement(ResultColumnList sourceRSRCL)
 
 				}else{
 					if(cd.isAutoincAlways()){
+						GemFireStore ms = Misc.getMemStore();
+						if (ms != null && ms.isSnappyStore() && sourceRSRCL.size() == size) {
```

Comment:
getMemStore() will never be null and will throw a CachedClosedException if null, so remove the null check.